web/concrete/src/Service/Configuration/HTTP/ApacheGenerator.php

```diff
@@ -0,0 +1,110 @@
+<?php
+namespace Concrete\Core\Service\Configuration\HTTP;
+
+use Concrete\Core\Service\Configuration\GeneratorInterface;
+use Concrete\Core\Application\Application;
+
+class ApacheGenerator implements GeneratorInterface
+{
+    /**
+     * @var Application
+     */
+    protected $app;
+
+    /**
+     * @var array
+     */
+    protected $rules;
+
+    /**
+     * @var array
+     */
+    protected $enabledRules;
+
+    /**
+     * Initializes the instance.
+     *
+     * @param Application $app
+     */
+    public function __construct(Application $app)
+    {
+        $DIR_REL = DIR_REL;
+        $DISPATCHER_FILENAME = DISPATCHER_FILENAME;
+        $this->app = $app;
+        $this->rules = array();
+        $this->enabledRules = array();
+        $this->addRule(
+            'pretty_urls',
+            array(
+                'commentBefore' => '# -- concrete5 urls start --',
+                'code' => <<<EOT
+<IfModule mod_rewrite.c>
+    RewriteEngine On
+    RewriteBase $DIR_REL/
+    RewriteCond %{REQUEST_FILENAME} !-f
+    RewriteCond %{REQUEST_FILENAME}/index.html !-f
+    RewriteCond %{REQUEST_FILENAME}/index.php !-f
+    RewriteRule . $DISPATCHER_FILENAME [L]
+</IfModule>
+EOT
+                ,
+                'commentAfter' => "# -- concrete5 urls end --",
+            ),
+            function (Application $app) {
+                (bool) $app->make('config')->get('concrete.seo.url_rewriting');
```

Comment:
The pretty url can be enabled (ie should be added to configuration) or disabled (ie should be removed from configuration)